pkg/controllers/state/node.go

```diff
@@ -53,17 +62,49 @@ func (c *NodeController) Reconcile(ctx context.Context, req reconcile.Request) (
 		}
 		return reconcile.Result{}, err
 	}
+
 	if err := c.cluster.updateNode(ctx, node); err != nil {
 		return reconcile.Result{}, err
 	}
-	// ensure it's aware of any nodes we discover, this is a no-op if the node is already known to our cluster state
+
+	if err := c.record(ctx, node); err != nil {
```

Comment:
Can this just be folded into `updateNode` and possibly dispatch to two different functions there if it's unwieldy?